tests/test_roc.py

```diff
@@ -161,3 +175,201 @@ def test_fpr_and_tpr_different_types(fpr, tpr):
+def test_plot_roc_valid_input_binary_like_array(y_test, y_score):
+    plot.roc(y_test, y_score)
+    plot.ROC.from_raw_data(y_test, y_score)
+
+
+@image_comparison(baseline_images=["roc_multi"])
+def test_roc_from_raw_data_y_test_binary(roc_multi_classification_raw_data):
+    y_test, y_score = roc_multi_classification_raw_data
+
+    y_test_bin = np.zeros((y_test.size, y_test.max() + 1))
+    y_test_bin[np.arange(y_test.size), y_test] = 1
+    plot.ROC.from_raw_data(y_test_bin, y_score)
+
+
+@image_comparison(baseline_images=["roc", "roc"])
+def test_roc_from_raw_data(y):
+    y_test, y_score = y
+    # roc_old_api
+    plot.roc(y_test, y_score)
+
+    plot.ROC.from_raw_data(y_test, y_score)
+
+
+@image_comparison(baseline_images=["roc", "roc"])
+def test_roc_y_score_vector(y):
+    y_test, y_score = y
+    y_score_vector = np.array([0.5, 0.6, 0.7, 0.8, 0.9, 0.9, 0.8, 0.1, 0.1, 0.3])
+
+    # roc_old_api
+    plot.roc(y_test, y_score_vector)
+
+    plot.ROC.from_raw_data(y_test, y_score)
+
+
+@image_comparison(baseline_images=["roc", "roc"])
+def test_roc(roc_values):
+    fpr, tpr = roc_values
+
+    # test when list is given
+    roc = plot.ROC(fpr, tpr)
+    roc.plot()
+
+    # test when np.array
+    roc = plot.ROC(np.array(fpr), np.array(tpr))
+    roc.plot()
+
+
+@image_comparison(baseline_images=["roc_multi"])
+def test_roc_multi_from_raw_data(roc_multi_classification_raw_data):
+    y_test, y_score = roc_multi_classification_raw_data
+    plot.ROC.from_raw_data(y_test, y_score)
+
+
+@image_comparison(baseline_images=["roc_multi_with_labels"])
+def test_roc_multi_from_raw_data_with_labels(roc_multi_classification_raw_data_set2):
+    y_test, y_score = roc_multi_classification_raw_data_set2
+    plot.ROC.from_raw_data(y_test, y_score)
+
+
+@image_comparison(baseline_images=["roc_multi", "roc_multi"])
+def test_roc_multi(roc_multi_classification_values):
+    fpr, tpr, label = roc_multi_classification_values
+    # test when fpr and tpr are lists
+    multi_roc = plot.ROC(fpr, tpr, label=label)
+    multi_roc.plot()
+
+    # test when fpr and tpr are np.arrays
+    fpr_np_array = [np.array(fpr_list) for fpr_list in fpr]
+    tpr_np_array = [np.array(tpr_list) for tpr_list in tpr]
+    multi_roc = plot.ROC(fpr_np_array, tpr_np_array, label=label)
+    multi_roc.plot()
+
+
+@image_comparison(baseline_images=["roc", "roc2", "roc_add_roc"])
+def test_roc_add_to_roc_from_raw_data(y):
+    y_test, y_score = y
+    y_test_roc2 = np.array([1, 1, 1, 1, 1, 0, 0, 0, 0, 0])
+    y_score_roc2 = np.array(
+        [
+            [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+            [0.3, 0.6, 0.7, 0.8, 0.9, 0.9, 0.8, 0.2, 0.3, 0.5],
+        ]
+    ).T
+
+    roc1 = plot.ROC.from_raw_data(y_test, y_score)
+    roc2 = plot.ROC.from_raw_data(y_test_roc2, y_score_roc2)
+    roc1 + roc2
+
+
+@image_comparison(baseline_images=["roc_add_roc"])
+def test_roc_add_to_roc(roc_values):
+    fpr1, tpr1 = roc_values
```

Comment:
Why do we need this? it seems like those values get overwritten?
I see similar stuff in other test cases as well

Reply:
If we can parameterize the inputs I think we should do it as the test seems pretty much the same for the roc add

Reply:
Good point. This was from the previous code where roc tests were scattered in different places. I removed all roc resources from `conftest.py` to `test_roc.py` and parameterized the "short" values that are easy to use. The more complex values I kept as a fixture.